pkg/cluster/controllers/database/readiness.go

```diff
@@ -106,22 +102,27 @@ func getDatabasePasswordRefName(name string) string {
 }
 
 // DeployComponentSecret deploy harbor component database secret.
-func (p *PostgreSQLController) DeployComponentSecret(ctx context.Context, conn *Connect, ns string, secretName string) (*corev1.Secret, error) {
+func (p *PostgreSQLController) DeployComponentSecret(ctx context.Context, conn *Connect, ns string, secretName string,
+	harborcluster *goharborv1.HarborCluster) error {
+
 	secret := &corev1.Secret{}
 	sc := p.GetDatabaseSecret(conn, ns, secretName)
+	if err := controllerutil.SetControllerReference(harborcluster, sc, p.Scheme); err != nil {
```

Comment:
Why let the cluster own the secret? The cluster has owned the psql CR, so is it possible to let the psql CR own the secret?

Reply:
This secret is store pg conn info for notary and core component, so harborcluster should own this secret.

Reply:
👌